ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/io/BigQuery.java

```diff
@@ -0,0 +1,124 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package com.mozilla.telemetry.ingestion.io;
+
+import com.google.cloud.bigquery.BigQueryError;
+import com.google.cloud.bigquery.InsertAllRequest;
+import com.google.cloud.bigquery.InsertAllResponse;
+import com.google.cloud.bigquery.TableId;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+
+public class BigQuery {
+
+  public static class Write implements Function<Write.TableRow, CompletableFuture<Write.TableRow>> {
+
+    private final com.google.cloud.bigquery.BigQuery bigquery;
+    private final Map<TableId, Batch> batches = new HashMap<>();
```

Comment:
This should probably be a ConcurrentHashMap if it's being accessed and mutated by multiple threads.